cost/state.go

```diff
@@ -37,17 +43,21 @@ func NewState(ctx context.Context, backend Backend, queries []query.Resource) (*
 		for _, comp := range res.Components {
 			prods, err := backend.Product().Filter(ctx, comp.ProductFilter)
 			if err != nil {
-				return nil, err
+				state.addComponent(res.Address, comp.Name, Component{Error: err})
+				continue
 			}
 			if len(prods) < 1 {
-				return nil, fmt.Errorf("product not found")
+				state.addComponent(res.Address, comp.Name, Component{Error: ErrProductNotFound})
+				continue
```

Comment:
I see that this is not validated anywhere no? Before it was stopping and not it continues but does not report the `Error` no?

I'm missing something maybe.

Reply:
The error is stored in the resource with `addComponent` (in the `Error` property of `Component`). The library user can then iterate over the resources and check the `Error` for each of them separately so that if one resource fails, the result is still returned.